contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,751 @@
+        // note: Although this does reduce usage of Ethereum state, it does not
+        // reduce the gas costs of removal transactions. We believe this is the
+        // best behavior at the time of writing, and do not expect this to
+        // result in any breaking changes in the future. All current proposals
+        // to change the self-destruct opcode are backwards compatible, but may
+        // result in not removing the bytecode from the blockchain state. This
+        // implementation is compatible with that architecture, as it does not
+        // rely on the bytecode being removed from the blockchain state.
+        dependencyType
+            .scriptBytecodeAddresses[dependencyType.scriptCount - 1]
+            .purgeBytecode();
+        // delete reference to contract address that no longer exists
+        delete dependencyType.scriptBytecodeAddresses[
+            dependencyType.scriptCount - 1
+        ];
+        unchecked {
+            dependencyType.scriptCount = dependencyType.scriptCount - 1;
+        }
+
+        emit DependencyTypeScriptUpdated(_dependencyTypeId);
+    }
+    
+    /**
+     * @notice Updates preferred CDN for dependency type `_dependencyTypeId`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _preferredCDN URL for preferred CDN.
+     */
+    function updateDependencyTypePreferredCDN(
+        bytes32 _dependencyTypeId,
+        string memory _preferredCDN
+    )
+        external
+        onlyAdminACL(this.updateDependencyTypePreferredCDN.selector)
+        onlyNonEmptyString(_preferredCDN)
+    {
+        dependencyTypeInfo[_dependencyTypeId].preferredCDN = _preferredCDN;
+
+        emit DependencyTypePreferredCDNUpdated(_dependencyTypeId, _preferredCDN);
+    }
+
+    /**
+     * @notice Updates preferred repository for dependency type `_dependencyTypeId`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _preferredRepository URL for preferred repository.
+     */
+    function updateDependencyTypePreferredRepository(
+        bytes32 _dependencyTypeId,
+        string memory _preferredRepository
+    )
+        external
+        onlyAdminACL(this.updateDependencyTypePreferredRepository.selector)
+        onlyNonEmptyString(_preferredRepository)
+    {
+        dependencyTypeInfo[_dependencyTypeId].preferredRepository = _preferredRepository;
+
+        emit DependencyTypePreferredRepositoryUpdated(
+            _dependencyTypeId,
+            _preferredRepository
+        );
+    }
+
+    /**
+     * @notice Updates project website for dependency type `_dependencyTypeId`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _projectWebsite URL for project website.
+     */
+    function updateDependencyTypeProjectWebsite(
+        bytes32 _dependencyTypeId,
+        string memory _projectWebsite
+    ) external onlyAdminACL(this.updateDependencyTypeProjectWebsite.selector) {
```

Comment:
needs `onlyExistingDependencyType` modifier.
are we okay having an empty string as input here?

Reply:
Good question, I could imagine a website goes down so it's no longer relevant but here's nothing new to point at

Reply:
> are we okay having an empty string as input here?

Yeah to @yoshiwarab's point I think allowing empty string probably makes sense, but worth clarifying that this is intentional in the doc comment for this method.